code/modules/admin/verbs/playsound.dm

```diff
@@ -48,6 +48,38 @@ var/list/sounds_cache = list()
 	play_sound(melody)
 	feedback_add_details("admin_verb","PSS") //If you are copy-pasting this, ensure the 2nd paramter is unique to the new proc!
 
+/client/proc/play_intercomm_sound()
+	set category = "Event"
+	set name = "Play Sound via Intercomms"
+	set desc = "Plays a sound at every intercomm on the station z level. Works best with small sounds."
+	if(!check_rights(R_SOUNDS))	return
+
+	var/A = alert("This will play a sound at every intercomm on the station Z, are you sure you want to continue? This works best with short sounds, beware.","Warning","Yep","Nope")
+	if(A != "Yep")	return
+
+	var/list/sounds = file2list("sound/serversound_list.txt");
+	sounds += "--CANCEL--"
+	sounds += sounds_cache
+
+	var/melody = input("Select a sound from the server to play", "Server sound list", "--CANCEL--") in sounds
+
+	if(melody == "--CANCEL--")	return

```

Comment:
Eugh, this is not how you handle canceling out of something. Give the list box a proper cancel button. And yes, I know you just copy-pasted this from `play_server_sound` - that isn't an excuse, because it also does it wrong.